src/index.ts

```diff
@@ -1,12 +1,173 @@
-import EnsemblGenomeBrowser from './ensemblGenomeBrowser';
+import { 
+  IncomingAction, IncomingActionType, OutgoingAction, OutgoingActionType, Subscribe
+} from './types';
 
-import { OutgoingAction, OutgoingActionType, IncomingAction, IncomingActionType } from './action';
+const subscriptions = new Map<IncomingActionType, Set<(action: any) => void>>();
 
-export {
-    OutgoingAction, 
-    OutgoingActionType, 
-    IncomingAction, 
-    IncomingActionType
+type GenomeBrowserType = {
+  go: (config_object: any) => void;
+  set_stick: (stickId: string) => void;
+  wait: () => void;
+  goto: (left: number, right: number) => void;
+  jump: (location: string) => void;
+  set_y: (y: number) => void;
+  set_switch: (path: string[]) => void;
+  clear_switch: (path: string[]) => void;
+  set_message_reporter: (callback: (...action: [type: IncomingActionType, payload: any]) => void) => void;
+};
+
+
+type ConfigData = {
+  backend_url?: string;
+  target_element_id?: string;
+}
+class EnsemblGenomeBrowser {
+  
+  genomeBrowser: GenomeBrowserType | null = null;
+  inited = false;
+
+  public async init(config: ConfigData = {}) {
+
+    if(!this.inited) {
+      const { default: init, GenomeBrowser } = await import('./peregrine/peregrine_ensembl.js');
+      await init();
+      this.genomeBrowser = new GenomeBrowser();
+      this.genomeBrowser?.go(config);
+    }
+    this.inited = true;
+    
+    this.genomeBrowser?.set_message_reporter(this.handleIncoming);
+    this.genomeBrowser?.set_stick("");
```

Comment:
Empty stick confuses the browser and is the cause of the long-lasting blank screen while the browser is trying to recover. Also, as Dan says, such a reset of the browser is unnecessary. This line should be deleted.